wall-e/skynet/drivers/motor.c

```diff
@@ -1,14 +1,101 @@
-/*
- * motor.c
- *
- *  Created on: Sep 29, 2021
- *      Author: faheem
- */
+//PWM Demo Code
+
+//ECE 4437
+
+
+//Include Headers
 
 #include "motor.h"
 
-void Motor_Init() {
-    // code here
+
+void Motor_Init(){
+    uint32_t val_load, pwm_clk;
+    uint16_t duty = 70;
+
+    SysCtlPeripheralEnable(SYSCTL_PERIPH_PWM1);
+
+
+    SysCtlPWMClockSet(SYSCTL_PWMDIV_64);
+
+    SysCtlPeripheralEnable(SYSCTL_PERIPH_PWM1);
+    SysCtlPeripheralEnable(SYSCTL_PERIPH_GPIOA);
+
+    SysCtlPeripheralEnable(SYSCTL_PERIPH_GPIOC);
+    GPIOPinTypeGPIOOutput(GPIO_PORTC_BASE, GPIO_PIN_7);
+    GPIOPinTypeGPIOOutput(GPIO_PORTC_BASE, GPIO_PIN_6);
+    GPIOPinWrite(GPIO_PORTC_BASE, GPIO_PIN_7, 0);
+    GPIOPinWrite(GPIO_PORTC_BASE, GPIO_PIN_6, 0);
+
+
+    GPIOPinTypePWM(GPIO_PORTF_BASE, GPIO_PIN_3);
+    GPIOPinTypePWM(GPIO_PORTF_BASE, GPIO_PIN_2);
+    GPIOPinConfigure(GPIO_PF3_M1PWM7);
+    GPIOPinConfigure(GPIO_PF2_M1PWM6);
+
+    pwm_clk = SysCtlClockGet() / 64;
+    val_load = (pwm_clk / 100) - 1;
+    PWMGenConfigure(PWM1_BASE, PWM_GEN_3, PWM_GEN_MODE_DOWN);
+    PWMGenPeriodSet(PWM1_BASE, PWM_GEN_3, val_load);
+    PWMGenConfigure(PWM1_BASE, PWM_GEN_1, PWM_GEN_MODE_DOWN);
+    PWMGenPeriodSet(PWM1_BASE, PWM_GEN_1, val_load);
```

Comment:
One of these sets of two lines probably can be removed. I think it's the pair that references `GEN_3` but try either and see which one works.